File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [0] * n
    
    def dfs(computers, visited, start):
        stack = [start]
        while stack:
            now = stack.pop()
            if visited[now] == 0:
                visited[now] = 1
            
            for i in range(len(computers)):
                if computers[now][i] == 1 and visited[i] == 0:
                    stack.append(i)
    
    current_index = 0
    while 0 in visited:
        if visited[current_index] == 0:
            dfs(computers, visited, current_index)
            answer += 1
        current_index += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                answer++;
                dfs(i, computers);
            }
        }
        
        return answer;
    }
    
    public void dfs(int idx, int[][] computers) {
        visited[idx] = true;
        
        for (int i = 0; i < computers[idx].length; i++) {
            if (computers[idx][i] == 1 & !visited[i]) {
                dfs(i, computers);
            }
        }
    }
}